### PR TITLE
Implement the `SleepPageView` and corresponding logic

### LINKED ISSUE #45: `SleepPageView`: сreate a screen and implement logic when `SleepButton` pressed and changing  to `SleepSetupView ` to `DuringSleepView`
### Depend on:
#22 #23 #24 

### Scope of the issue (per agreement with the team on May, 23th)
- [x] #69
- [x] #71
- [x] #75
- [x] #81
- [x] #80
- [x] #85 (to show on the SleepPageView)
- [ ] Implement the current task and logic to handle persistence. Specifically:
  - create the `SleepPageView` view (assemble from the existing views)
  - create the `SleepPageViewModel`

### DoD:
Create a `SleepPageView` screen as shown in [Figma](https://www.figma.com/file/Q3Vq556r3LTQrOokAggk1Z/Sleep-Magic-App-Design?node-id=0%3A1&t=9Jf97CaFnqWDDYri-1)
#### When the user clicks on `StartSleeping` button:
- `SleepSetupView` is hidden and `DuringSleepView` is displayed (no animation)
- text on `StartSleeping` is changed from `Start sleeping` to `Stop Sleeping`

#### On `DuringSleepView`:
- if the alarm was on - `alarmLabel` shows the alarm time in `hh:mm` format
- if the alarm was off - `alarmLabel` shows `no alarm` text

#### What view to show:
- If the user closed the app when the sleeping mode was on and reopens it again - the app should display the `DuringSleepView`
- If the user closed the app in nonSleeping mode  - the app should display the `SleepSetupView`

#### Details:
- implement passing of information from SleepSetupView to DuringSleepView using MVVM
- when pressing the `StratSleeping` button the app should save the current time and Sleep mode in `UserDefaults`, if the user went to sleep, the mode is `true` ( think about keeping the time zone)
- save time pressing the button in `UserDefaults`
- save a boolean value of `isSleep`, if the user pressed `Start Sleeping` save `true` in `UserDefaults`
- implement logic on how to show screens if the user killed the application, if `isSleep = true` show `DuringSleepView` and if `isSleep` = `false` - `SleepSetupView`

### PR DESCRIPTION
**Pull request by dersim-davaod**
Resolves #45

### CODE REVIEW

File: Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import Foundation
+
+final class SleepPageViewModel: ObservableObject {
```
Comment (MikhailKosarev) [inline Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift:3]: should we add documentation for the class?

File: Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+        let totalDifferenceInMinutes = nowTotalMinutes < alarmTotalMinutes
+            ? alarmTotalMinutes - nowTotalMinutes
+            : (hoursInDay * minutesInHour) - nowTotalMinutes + alarmTotalMinutes
+
+        let hours = totalDifferenceInMinutes / minutesInHour
+        let minutes = totalDifferenceInMinutes % minutesInHour
+
+        estimatedSleepTimeText = isAlarmOn
+            ? "\(estimatedSleepTimePrefixText):" +
```
Comment (MikhailKosarev) [inline Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift:74]: a little hard to read. Maybe smth like this:
```suggestion
            ? "\(estimatedSleepTimePrefixText):" + "\n" +
            "\(hours)h" + " " + "\(minutes)m"
            : estimatedSleepTimeNotAvailableText
```
?

**Review by MikhailKosarev — CHANGES_REQUESTED** [review]
have some questions:

File: Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift (hunk 2 above)
Comment (dersim-davaod) [reply Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift]: Yeah, the linter breaks "beautiful" formatting.
I've updated it, please check it out.

File: Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift (hunk 1 above)
Comment (dersim-davaod) [reply Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift:4]: Sure, updated!

File: Sleep Tracker/SleepTracker/View/SleepPageView/DuringSleepView/DuringSleepView.swift [hunk 3]
```diff
@@ -4,6 +4,9 @@ import SwiftUI
 struct DuringSleepView: View {
     // MARK: - Non-private interface
 
+    /// Text to show whether the alarm on/off. If the alarm. If the alarm is off,
```
[the remark this answers is not in the document]
Comment (dersim-davaod) [reply Sleep Tracker/SleepTracker/View/SleepPageView/DuringSleepView/DuringSleepView.swift]: Fixed.

**Review by marinkman — COMMENTED** [review]
Generally good, but `Predicted sleep duration` should be from 0m to 23h 59m.
It should be 0m at the current time, not 24h 0m.

**Review by MikhailKosarev — CHANGES_REQUESTED** [review]
Overall, it looks good. There is only one question:

File: Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,151 @@
+import Foundation
+
+/// Manages UI of the `SleepPageView` and contains a related business logic.
+final class SleepPageViewModel: ObservableObject {
```
[the remark this answers is not in the document]
Comment (dersim-davaod) [reply Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift:4]: @MikhailKosarev yes, the logic in the `SleepPageViewModel`.
Specifically, the `isSleepInProgress` property contains it.

Does it work incorrectly?

**Review by marinkman — APPROVED** [review]
LGTM

File: Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (MikhailKosarev) [reply Sleep Tracker/SleepTracker/ViewModel/SleepPageViewModel.swift:4]: Everything is ok

**Comment by dersim-davaod** [comment]
The `CHANGELOG.md` has been updated.

**Review by MikhailKosarev — APPROVED** [review]
LGTM

**Pull request merged**